examples/spark2/src/main/scala/org/apache/carbondata/examples/S3UsingSDkExample.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.examples
+
+import org.apache.hadoop.fs.s3a.Constants.{ACCESS_KEY, ENDPOINT, SECRET_KEY}
+import org.apache.spark.sql.SparkSession
+import org.slf4j.{Logger, LoggerFactory}
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.sdk.file.{CarbonWriter, Schema}
+
+object S3UsingSDKExample {
+
+  // prepare SDK writer output
+  def buildTestData(
+      path: String,
+      num: Int = 3,
+      persistSchema: Boolean = false): Any = {
+
+    // getCanonicalPath gives path with \, so code expects /. Need to handle in code ?
+    val writerPath = path.replace("\\", "/");
+
+    val schema = new StringBuilder()
```

Comment:
To make it simpler, can you use the object way to construct the schema object

Reply:
I refer merged code in carbondata, there are many usage like that: 

org.apache.carbondata.spark.testsuite.createTable.TestCarbonFileInputFormatWithExternalCarbonTable

org.apache.carbondata.spark.testsuite.createTable.TestUnmanagedCarbonTable

Keep it? or change all similar place in carbondata?

Reply:
please refer to CSVCarbonWriterTest.testWriteFiles

Reply:
ok,done